src/SoapCore/Meta/MetaBodyWriter.cs

```diff
@@ -462,10 +462,27 @@ private void AddTypes(XmlDictionaryWriter writer)
 					writer.WriteStartElement("restriction", Namespaces.XMLNS_XSD);
 					writer.WriteAttributeString("base", $"{_xmlNamespaceManager.LookupPrefix(Namespaces.XMLNS_XSD)}:string");
 
-					foreach (var value in Enum.GetValues(toBuild))
+					var membersWithCustomNames = toBuild.GetMembersWithAttribute<XmlEnumAttribute>();
+					foreach (var value in Enum.GetNames(toBuild))
 					{
+						var xmlEnumAttribute = membersWithCustomNames
+												.FirstOrDefault(m => m.Name == value)?
+												.CustomAttributes
+												.FirstOrDefault(c => c.AttributeType == typeof(XmlEnumAttribute));
+						var nameFromAttribute = (xmlEnumAttribute?
+													.ConstructorArguments?
+													.FirstOrDefault()
+													.Value
+												??
+												xmlEnumAttribute?
+													.NamedArguments?
+													.FirstOrDefault(a => a.MemberName == "Name")
+													.TypedValue
+													.Value)?
+												.ToString();
+

```

Comment:
This code traverses all members with custom names for every member resulting in a potential O(n^2) perf characteristic (since it traverses all members with custom attribute for every member in the enum)

Please see if you can improve that. I gave it a shot, and my approach is included below. Feel free to use that implementation if you want to.

I tried optimizing the code with an approach like this
```
Type toBuild = typeof(MyEnum);
var membersWithCustomNames = from n in Enum.GetNames(toBuild)
							 join m in toBuild.GetMembers() on n equals m.Name
							 let ca = m.CustomAttributes.FirstOrDefault(ca => ca.AttributeType == typeof(XmlEnumAttribute))
							 select new
							 {
								 Name = m.Name,
								 CustomName =
								 (ca?
												.ConstructorArguments?
												.FirstOrDefault()
												.Value
											??
											ca?
												.NamedArguments?
												.FirstOrDefault(a => a.MemberName == "Name")
												.TypedValue
												.Value)?
											.ToString()
							 };
foreach (var value in membersWithCustomNames)
{
	var finalValue = value.CustomName ?? value.Name;
}
```

Results from BenchmarkDotnet (my implementation being the second one)
![image](https://github.com/DigDes/SoapCore/assets/516834/89fbc2e1-e78c-4391-973b-befc80b1c48c)